kernel/hpet.c

```diff
@@ -48,15 +48,15 @@ void hpet_reset_timeout (void) {
 
 LIST_HEAD(sleep_queue);
 
-void hpet_nanosleep (void) {
-  uint64_t nanosecs = running_tcb->saved_registers->rbx;
+void __attribute__((noreturn)) hpet_nanosleep (void) {
+  uint64_t nanosecs = running_tcb->saved_registers.rbx;
   uint64_t femtosecs = nanosecs * 1000000;
   uint64_t ticks = femtosecs / (hpet_reg->capabilities >> 32);
   uint64_t cur = hpet_reg->main_counter;
   uint64_t deadline = cur + ticks;
   running_tcb->wakeup_deadline = deadline;
   if (deadline < cur) {
-    return;
+    return_to_current_thread();
```

Comment:
That's true, yeah, it doesn't follow that rule.  I'm... still conflicted on the rule anyways, and since this is just about the safest possible case [an early guard check before any operation that modifies pretty much anything], I think I'm okay with leaving it like this.  In the future we can experiment and see what style is easiest to reason about.

That said, I will move the check before the line right before it, just to emphasize that we don't modify /anything/ before that check.